blueocean-jira/src/main/java/io/jenkins/blueocean/service/embedded/jira/JiraSCMListener.java

```diff
@@ -40,38 +47,53 @@ public void onChangeLogParsed(Run<?, ?> run, SCM scm, TaskListener listener, Cha
             return;
         }
         // Query for JIRA issues
-        Set<JiraIssue> issuesFromJqlSearch = Sets.newHashSet(Iterables.transform(session.getIssuesFromJqlSearch(jql), new Function<Issue, JiraIssue>() {
-            @Override
-            public JiraIssue apply(Issue input) {
-                return new JiraIssue(input);
+        try {
+            List<Issue> issues = session.getIssuesFromJqlSearch(jql);
+            Set<JiraIssue> issuesFromJqlSearch = issues == null ? Collections.emptySet() :
+                issues.stream().map( input -> new JiraIssue(input) )
+                    .collect( Collectors.toSet() );
+
+            // If there are no JIRA issues, do not update the actions
+            if (issuesFromJqlSearch.isEmpty()) {
+                return;
             }
-        }));
-        // If there are no JIRA issues, do not update the actions
-        if (issuesFromJqlSearch.isEmpty()) {
-            return;
-        }
-        // Create or update the JiraBuildAction
-        JiraBuildAction action = run.getAction(JiraBuildAction.class);
-        if (action == null) {
-            run.addAction(new JiraBuildAction(run, issuesFromJqlSearch));
-        } else {
-            action.addIssues(issuesFromJqlSearch);
+            // Create or update the JiraBuildAction
+            JiraBuildAction action = run.getAction(JiraBuildAction.class);
+            if (action == null) {
+                run.addAction(new JiraBuildAction(run, issuesFromJqlSearch));
+            } else {
+                action.addIssues(issuesFromJqlSearch);
+            }
+            run.save();
+        } catch (Exception e ){ // we do not want to fail the build if an issue happen here
+            LOGGER.warn( "skip reccording associated jira issues: {}", e.getMessage() );
+            // stack trace in debug mode
+            LOGGER.debug( e.getMessage(), e);
         }
-        run.save();
     }
 
-    static String constructJQLQuery(List<String> issueKeys) {
+    static String constructJQLQuery( Collection<String> issueKeys) {
         StringBuilder jql = new StringBuilder();
         jql.append("key in (");
-        for (int i = 0; i < issueKeys.size(); i++) {
+        Iterator<String> iterator = issueKeys.iterator();
+        while ( iterator.hasNext() ) {
+            String key = iterator.next();
             jql.append("'");
-            jql.append(issueKeys.get(i));
+            jql.append(key);
             jql.append("'");
-            if (issueKeys.size() > 1 &&  i+1 < issueKeys.size()) {
+            if (iterator.hasNext()) {
                 jql.append(",");
             }
         }
         jql.append(")");
         return jql.toString();
     }
+
+    static Collection<String> getIssueKeys(ChangeLogSet<?> changelog, Pattern issuePattern) {
+        Set<String> issueKeys = new HashSet<>();
```

Comment:
Nice one @olamy, removal of duplicate issues, this should improve things overall 👏